### PR TITLE
Recreate blocks using only wp core - remove all css + Min WP Version Bump to 5.8

### PR DESCRIPTION
**Pull request by godleman**
### Description of the Change

Aims of these changes:
WP patterns cannot contain CSS. We need to refactor using only wp core settings in order to prepare to add patterns to WP pattern library.

Actions from this change:

- Recreating the blocks using current WP core block options only. 
- Replacing old blocks with wp-core generated source code. 
- Removing all block css in favour of inlined styling from wp core settings.
- Replacing all images with public urls from openverse
- Also bumping the WP minimum required version from `5.7` to `5.8` because, after this PR, the block patterns will be broken in 5.7 or below.

Scope outlined by @jeffpaul 
See @jeffpaul for documentation if needed.

Closes #

### How to test the Change

Create the patterns in the plugin in wp and observe the output in the editor and FE. 

### Changelog Entry

> Added - WP Core only block code
> Changed - Existing block code and css
> Deprecated - n/a
> Removed - Existing block code and css
> Fixed - n/a
> Security - n/a

### Notes / Issues in recreating design using only wp core blocks settings
**Global content width**
No option change, so the narrow 650px content width restricts the design. For instance font sizes need to be decrease otherwise they'll look oversized against the narrow content width

**Group blocks** 
You can add borders but you can't limit the border stroke to only the bottom of the element - it's always a full element border. You can only control the border width and style. 

**Text elements** 
Margin - you can add positive margin values but you cannot reset core text element margins or apply negative margin values to compensate. 

**Tables**
Stripes - you can add stipes but you can't control the stripe background color value - its always grey 

**Tabs**
Heading - There is a heading expectation when you create a Tabs Block. But this heading does not have the usual Heading element controls like font-size, appearence etc. So it is best to create a core Heading element above the tabs block. But the Tabs block blank heading still renders and adds unwanted margin top and bottom. 

**Tabs Item** 
There are not styling controls on Tab Items - like modifying or removing the default borders or modifying the Tab heading padding to match a design 

**Columns** 
Style - no option to add border or box-shadow - so you can't replicate box Column box shadows from design

**Lists** - no option to replace standard HTML list item bullets with images/icons - so better to just use a 'Row' with an image and a paragraph block. 

**Row** - You can justify items but you can't align items - so you can't have a row with top aligned content, its always center aligned

### Credits

Props @username, @username2, ...

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Review by faisal-alvi — CHANGES_REQUESTED** [review]
@cldhmmr thank you for working on this. I have tested the PR. Please check the following points that I observed while testing:

1. The cover block pattern seems to be broken:

image

2. All blocks have the same title "2.5M",

image

it should be different for each block:

image

3. The active tab item font size is tiny, not sure if this is caused by the current implementation, but it would be great if it's fixed. Tested with the Storefront theme.

image

4. Our Packages section seems broken, looks like the bullet images are 404 not found.

image

5. Also the Media Kit page is not 100% responsive now, we can see the large fonts in the Tabs block and the CTA inline button. Should we keep it as it is? cc @jeffpaul 

image

image

6. The font sizes are different in the Audience Profile pattern:

image

7. The cypress test is [failing](https://github.com/10up/publisher-media-kit/actions/runs/3290571184/jobs/5423590603) for WP#5.7. Seems like the test can not find the `core/heading` element.

image

**Comment by godleman** [comment]
Thanks @faisal-alvi 
This wasn't happening locally, I'll look in to these and update asap

**Comment by godleman** [comment]
@faisal-alvi cc @jeffpaul  I've made some changes, please review my responses below. I think its best to recreate the blocks fresh to see the changes :

**The cover block pattern seems to be broken:**
I've changed the image url for this and the block seems ok. 
image

**All blocks have the same title "2.5M":**
Updated to variations thanks 

**The active tab item font size is tiny, not sure if this is caused by the current implementation, but it would be great if it's fixed. Tested with the Storefront theme.**
I've updated this in the Tab block editor styles. 
image


**Our Packages section seems broken, looks like the bullet images are 404 not found.**
The images are now updated to public urls - they were referencing local, so this should be fixed 
image


**Also the Media Kit page is not 100% responsive now, we can see the large fonts in the Tabs block and the CTA inline button. Should we keep it as it is? cc @jeffpaul**

I haven't been able to replicate an issue with the table itself - the table show a horizontal overflow scrollbar on mobile and the user can slide that. Could you verify please?
image

I agree with the Still Have Questions CTA block, there is currently not a setting in wordpress core to control the layout on mobile, so we are stuck with the row alignment in this case on mobile.
image

**The font sizes are different in the Audience Profile pattern:**
This seemed to be a inconsistency in the output from editor - I think this is fixed now. I have also changed the images to be more like the original design using wiki commons images. I am a little bit hesitant about the images being outdated gender archetypes and would suggest we change to something non-gendered like animals or lego, or something else. Thoughts on this appreciated. 
image

**The cypress test is [failing](https://github.com/10up/publisher-media-kit/actions/runs/3290571184/jobs/5423590603) for WP#5.7. Seems like the test can not find the core/heading element.**

I don't understand why this is being thrown or how to fix this? I'm not sure if related to the updates or not. Please let me know if you have any ideas?

**Important note about images**
A lot of time is spent around images and images changes. 
Issues are :
Wordpress want us to use OpenVerse 
OpenVerse images are often sourced from personal Flickr accounts which we wouldn't really want to embed in our plugin
OpenVerse in general has a weak search facility and it can take a long time to find appropriate images
When logged into wordpress.org you can create a new pattern directly and the openverse image search is integrated, so searching here and using url from there seems like a good idea. But, there are 2 problems. a) the search facilty is terrible, even worse than the main openverse site, and b) the editor often does not like the image urls and they don't display. Example of terrible search below
image

OpenVerse however also sources images from wikipedia commons, which has a much better search capability 
I have updated the images to use image urls from wikipedia commons, using their no-restrictions filter. I am hoping that is ok for the publicly accessible URLs. They load fine in the editor and FE.

Thanks

**Review by faisal-alvi — CHANGES_REQUESTED** [review]
@cldhmmr thank you for working on this. The changes look good, however, I have observed a few things. Please check the following details.

1. The cover block seems okay but when I switched to the storefront theme, the text is not clearly visible, should we add a color so it's white regardless of the selected theme?

image

2. The text "Monthly Unique Views on 10up.com" should be "Monthly Page Views on 10up.com"

image

3. 
> I haven't been able to replicate an issue with the table itself - the table show a horizontal overflow scrollbar on mobile and the user can slide that. Could you verify please?

Yes, there is a horizontal bar. I am seeing a bit of big fonts in the `storefront` theme. Also, I have seen the variations in the font size.

For example, 23px in the "Digital Ad Specs":

image

And 16px in the "Our Rates":

image

However, if I check in the `twentytwentyone` theme, the font size is same in both sections.

image

image

4. 
> I am a little bit hesitant about the images being outdated gender archetypes and would suggest we change to something non-gendered like animals or lego, or something else. Thoughts on this appreciated.

@jeffpaul any thoughts on ^this point?

5. Cypress tests failing.
> I don't understand why this is being thrown or how to fix this? I'm not sure if related to the updates or not. Please let me know if you have any ideas?

I have checked in WordPress 5.7, the block patterns are broken.

image

This is the reason why the Cypress test is failing. Seem like the markup is different in the 5.7 and the latest version. Can you please check? I think we might have to bump a WP version. cc: @jeffpaul 

6. Thank you for the details on the image search. Just wanted to know if image URLs from wikipedia commons will be accepted when we try to submit these block patterns to the WordPress pattern library.

**Comment by jeffpaul** [comment]
@faisal-alvi sounds like it'll be best if you can help pick up those suggested updates to help move the PR along, please and thanks!

**Comment by faisal-alvi** [comment]
Note: Bumping the WP minimum required version from `5.7` to `5.8` because, after this PR, the block patterns will be broken in 5.7 or below.

**Comment by faisal-alvi** [comment]
@jeffpaul can you please request someone from the OSP team for a code review + QA testing? I've tried requesting `10up/open-source-practice` but it didn't re-requested anyone.

**Comment by jeffpaul** [comment]
@10up/open-source-practice somethings amiss on GH code reviews, so hoping one of you can chime in with review here... thanks!

File: assets/css/frontend/components/index.css [hunk 1]
```diff
@@ -1,9 +1 @@
 /* Components */
```
Comment (dkotter) [inline assets/css/frontend/components/index.css:1]: Any reason to keep this file anymore? Seems like we've gotten rid of all component files, wondering if we should just delete that directory all together?

**Comment by dkotter** [comment]
I've done an initial code review and things look good. Haven't had time to actually test yet though. Just one question: How does this impact anyone that's already using these patterns? Just want to ensure we don't push out an update that breaks any existing content created with the old patterns.

**Comment by peterwilsoncc** [comment]
> Just want to ensure we don't push out an update that breaks any existing content created with the old patterns.

To cover this situation, I suggest maintaining some JavaScript and using [the deprecation functionality of the block editor](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/). I think this is possible when migrating to different blocks but it is possibly quite fiddly.

**Comment by faisal-alvi** [comment]
> Just want to ensure we don't push out an update that breaks any existing content created with the old patterns.

Our plugin has only one custom block, the "Tabs" block, which is used in the block patterns. Other than that, everything else that is used in the block patterns is the core blocks. We have not changed anything in our custom block (Tabs block), so this PR will not affect any existing content created with old patterns. I've also tested it again and nothing is found broken.

### The `develop` branch with old block patterns:

image

### This PR branch with the same old block patterns:

image

**Comment by jeffpaul** [comment]
@faisal-alvi let's discuss the goals of this PR in our next 1:1 and determine the best path forward

**Comment by iamdharmesh** [comment]
For reference: 
- [Patterns visauls with `develop` branch](https://user-images.githubusercontent.com/10613171/210967481-ade84ac5-2fa0-4670-9814-edf6ec5f2ca9.png)
- [Pattens visauls with this branch](https://user-images.githubusercontent.com/10613171/210967505-b7cdea67-984b-45ab-8001-36fe6e21318c.png)

**Comment by faisal-alvi** [comment]
Update: Jeff and I spoke about not merging this PR into the plugin and keeping all the CSS to maintain the original design. In order to submit them to the [WordPress block pattern directory](https://wordpress.org/patterns/), the primary goal of this PR was to design a block pattern without CSS (because WP does not accept custom CSS).

The next step will be to work on creating draft patterns directly in the WP directory and if that meets our expectations, we will submit them to the WordPress directory. A FEE will be assigned for this work.

_Screenshot of the editor in the WP directory:_
image

The minimum WordPress version bump to 5.8 was one modification that could be preserved because block patterns were broken in that version and below. However, after testing, I discovered that block patterns are not broken in the `develop` branch, so we may leave that change in place as well.

Closing this PR.

**Pull request closed without merging**